pkg/orchestrator/interfaces.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package orchestrator
+
+import (
+	"context"
+	"errors"
+)
+
+// Reactor is a stateful transform of states.
+// It models Owner and Processor, which reacts according to updates in Etcd.
+type Reactor interface {
+	Tick(ctx context.Context, state ReactorState) (nextState ReactorState, err error)
+}
+
+var (
+	// ErrEtcdTryAgain is used by a PatchFunc to force a transaction abort.
+	ErrEtcdTryAgain = errors.New("ErrEtcdTryAgain")
+	// ErrEtcdIgnore is used by a PatchFunc to signal that the reactor no longer wishes to update Etcd.
+	ErrEtcdIgnore = errors.New("ErrEtcdIgnore")
+	// ErrReactorFinished is used by reactor to signal a **normal** exit.
+	ErrReactorFinished = errors.New("ErrReactorFinished")
```

Comment:
What about moving them to pkg/errors?

Reply:
My thoughts are that they are not real errors, but are status codes and they do not signal that something has gone wrong. Maybe I can use some other data type to express these status codes if there is confusion.